src/modules-load/modules-load.c

```diff
@@ -39,12 +45,25 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent) {
+static int send_module_to_load(const int sock, const char *module) {
+        int r;
+
+        assert(sock >= 0);
+        assert(module);
+
+        r = RET_NERRNO(send(sock, module, strlen(module), 0));
```

Comment:
send() returns ssize_t (i.e. typically 64bit) and not int (i.e. typically 32bit). This is hence a bit sloppy, because it drops the upper 32bit on the floor just like that.

Reply:
Reworked to account for `ssize_t`